prdoc/pr_5723.prdoc

```diff
@@ -0,0 +1,15 @@
+title: Adds `BlockNumberProvider` in multisig, proxy and nft pallets
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      This would allow these pallets to use the relay chain block number when deployed on a parachain. 
```

Comment:
What this PR in fact is doing is adding the ability for pallets to specify their source of block number.

Then, you should describe in which case this change is deployed backwards compatible: RC -> RC, or Para -> Para. 

Then, you should describe if otherwise (RC -> Para, Para -> RC), what migratiion should be used. 

@davidk-pt is trying to introduce such a migration in https://github.com/paritytech/polkadot-sdk/pull/5656, I suggest you two to work together, first polish the `type BlockNumberProvider`, then work on a reusable migration, then move individual pallets.

Reply:
Still is a relevant point.